aioresponses/core.py

```diff
@@ -156,7 +156,10 @@ def _build_response(self, url: 'Union[URL, str]',
             method=method,
             headers=CIMultiDictProxy(CIMultiDict(**request_headers)),
         )
-        kwargs['writer'] = Mock()
+        if AIOHTTP_VERSION >= parse_version("3.9.0b1"):
+            kwargs['writer'] = AsyncMock()()
+        else:
+            kwargs['writer'] = Mock()
```

Comment:
I thought this over a bit more, and you can probably just set it `None`, as that will then skip the awaits etc.

The code should also be the same on all versions of aiohttp, the old Mock() will obviously still fail if it hits an await, it's just luck that no code ever reached that. The reason you need to call it, is that it returns something mocking an async function, _writer is a task/coroutine that is returned from an async function. You could also just use anything that supports `__await__()`.

```suggestion
        kwargs['writer'] = None
```

Reply:
I have tested it and this solution works, I like it even more because this way we avoid that in case of internal changes, it will always be previously verified to validate that it is not None.